js/main.js

```diff
@@ -1,2 +1,44 @@
 /* global data */
 /* exported data */
+var $title = document.querySelector('#note-title');
+var $urlInput = document.querySelector('#url');
+var $notes = document.querySelector('#entry-notes');
+var $submit = document.querySelector('form');
+
+function setImage(event) {
+
+  document.querySelector('img').setAttribute('src', $urlInput.value);
+}
+
+$urlInput.addEventListener('blur', setImage);
+
+function resetForm(event) {
+  data.entries.push({
+    title: $title.value,
+    photoURL: $urlInput.value,
+    notes: $notes.value
+  });
+  data.nextEntryId += 1;
+
+  $title.value = '';
+  $urlInput.value = '';
+  $notes.value = '';
+  document.querySelector('img').setAttribute('src', 'images/placeholder-image-square.jpg');
+  event.preventDefault();
+
+}
+
+$submit.addEventListener('submit', resetForm);
+function getPreviousEntries(event) {
+  return localStorage.setItem('note-pages', JSON.stringify(data));
+}
```

Comment:
This should only be done before the page unloads, not after every time the form is submitted. Also, all `localStorage` code should be in `data.js`, not `main.js`